docs/rfc/0000-radicle_daemon.md

```diff
@@ -0,0 +1,303 @@
+for interacting with Radicle's family of `Identity` types. See
+[spec/identities][id] for more details.
+
+```rust
+pub trait read::Identity<I> {
+	type Error;
+	
+	fn get<R>(&self, urn: Urn<R>) -> Result<Option<I>, Self::Error>;
+	fn list<R>(&self) -> Result<impl Iterator<Item = Result<I, Self::Error>, Self::Error>;
+}
+
+pub trait write::Identity<I> {
+	type Error;
+	
+	fn create<P, D>(&self, payload: P, delegations: D) -> Result<I, Self::Error>;
+	fn update<P, D>(&self, urn: Urn<R>, payload: P, delegations: D) -> Result<I, Self::Error>;
+	fn merge<R>(&self, urn: Urn<R>, from: PeerId) -> Result<I, Self::Error>;
+}
+
+pub trait read::Rad<I> {
+	type Error;
+	type Person;
+	type Signatures;
+	
+	/// rad/self
+	fn whoami(&self, urn: Urn<R>, peer: Option<PeerId>) -> Result<Option<Self::Person>, Self::Error>;
+	
+	/// rad/signed_refs
+	fn signatures(&self, urn: Urn<R>, peer: Option<PeerId>) -> Result<Option<Self::Signatures>, Self::Error>;
+	
+	/// rad/ids/*
+	fn delegates(&self, urn: Urn<R>, peer: Option<PeerId>) -> Result<impl Iterator<Item = Result<Self::Person, Self::Error>>, Self::Error>;
+	
+	/// rad/ids/<id>
+	fn delegate(&self, urn: Urn<R>, peer: Option<PeerId>) -> Result<Option<Self::Person>, Self::Error>;
+}
+```
+
+The `I` parameters signify that the traits are open to many
+identities, and for now there would be specific implementations for
+`Person` and `Project`. We leave any domain specific types open using
+associated types on the trait, e.g. `Rad::Person = Person` will be
+associated type for a `Project`'s `rad/self`.
+
+### Tracking
+
+As well as the family of `Identity` capabilities, we will also want
+methods for the tracking graph of an identity, sketched below:
+
+```rust
+pub trait Tracking {
+	type Error;
+	type Tracked;
+	
+	fn track<R>(&self, urn: Urn<R>) -> Result<bool, Self::Error>;
+	fn untrack<R>(&self, urn: Urn<R>) -> Result<bool, Self::Error>;
+	fn tracked<R>(&self, urn: Urn<R>) -> Result<Tracked, Self::Error>;
+}
+```
+
+### Profile
+
+The profile capability defines how a person can create a profile, read
+the current profile, and switch between existing profiles. 
+
+```rust
+pub trait Profile {
+	type Error;
+	type LocalIdentity;
+	
+	fn create(&self, whomai: Self::LocalIdentity) -> Result<(), Self::Error>;
+	fn update(&self, whoami: Self::LocalIdentity) -> Result<(), Self::Error>;
+	fn current(&self) -> Result<Option<Self::LocalIdentity>, Self::Error>;
+	fn switch(&self, other: Self) -> Result<Self, Self::Error>;
+}
+```
+
+There is an open question of how `Profile` will interact with a
+backing implementation that needs to be aware about restarting a
+loop. For example, if the backing implementation is `net::Peer`, then
+it will need to restart the `Peer` so it can re-initialise with the
+new `Storage`.
+
+### Replication
+
+TODO
+
+```rust
+pub trait Replication {
+	type Error;
+	type Result;
+	
+	fn replicate(&self, /* TODO */) -> Result<Self::Result, Self::Error>;
+}
+```
+
+### Gossip
```

Comment:
I’m not sure we want to surface this as is — announcing is handled by the daemon internally, no? An API for querying providers of a given URN is probably useful, but note that the response is necessarily streaming.

Reply:
Ya, actually that's a good point. @xla and I can brainstorm on higher-level components that we want. For example, `ask`ing for a particular `Identity` could be one capability. This would be one level up from something like `providers`, and would allow us to try different implementations but keeping the same API. I think these things would be _morally_ gossip protocols since they'll rely on the network, so I might group them under a `Gossip` header -- if that makes sense to others?